drivers/include/si1133.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2020 iosabi
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    drivers_si1133 Si1133 UV Index/Ambient Light Sensor with I2C
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ * @brief       Device driver for the Si1133 sensor
+ *
+ * The Si1133 is a UV Index Sensor and Ambient Light Sensor with I2C digital
+ * interface and programmable-event interrupt output.
+ *
+ * The I2C protocol implemented in this driver is most similar in registers and
+ * commands to the Si115x family, like the SI1153, however the Si1133 supports
+ * UV index while the Si115x doesn't.
+ *
+ * This driver provides @ref drivers_saul capabilities as well.
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for the Si1133 sensor
+ *
+ * @author      iosabi <iosabi@protonmail.com>
+ */
+
+#ifndef SI1133_H
+#define SI1133_H
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Driver error return codes
+ */
+typedef enum {
+    SI1133_OK           = 0,    /**< No error. */
+    SI1133_ERR_PARAMS   = -1,   /**< Invalid parameters. */
+    SI1133_ERR_I2C      = -2,   /**< I2C communication error. */
+    SI1133_ERR_LOGIC    = -3,   /**< Device communication logic error. */
+    SI1133_ERR_NODEV    = -4,   /**< No SI1133 device detected. */
+} si1133_ret_code_t;
+
+/**
+ * @brief Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;              /**< I2C bus the sensor is connected to */
+    /**
+     * @brief sensor address.
+     * Note: it is possible to change the a sensor's I2C address after it has
+     * been initialized to have multiple sensor on the same bus, but this is not
+     * supported by this driver. However, two different addresses can be
+     * selected by hardware.
+     */
+    uint8_t address;
+} si1133_params_t;
+
+/**
+ * @brief Device descriptor for the Si1133 sensor
+ */
+typedef struct {
+    /* Initialization parameters */
+    i2c_t i2c_dev;              /**< I2C bus the sensor is connected to */
+    uint8_t address;            /**< sensor address */
+    /* Internal members */
+    uint8_t cmd_counter;        /**< Si1133 command counter */
+    uint8_t sensors_mask;       /**< Mask of configured sensors */
+    uint32_t parsed_values[6];  /**< Sensor values read back. */
```

Comment:
why are they stored here?
Can't we directly return them in `si1133_capture_sensors()`?

Reply:
The order of the channels is selected by the driver (it happens to be sorted by smaller bits first), so if you pass 1|2|4 as sensors_mask you will get channel[0] with sensor 1<<0, channel[1] with sensor 1<<1, channel[2] with sensor 1<<2... but a different mask would give you the sensors in different order. What API do you propose to use to get the channel values in the same call to `si1133_capture_sensors()`.

The other less important reason in this commit is that when we switch to use the non-blocking mode (get an interrupt in a GPIO when the samples are ready / periodically) I should also have a place where to store these and a way for the user to get them, although at that time maybe they can be passed as parameters to the user callback too. 

Sensor values can get more complicated quickly, you can select different settings (different gain or different sample rate for periodic updates).

I can change this to a different API where you expose the 6 channel setup to the user and let the user map sensors to channels, this would use first one of these functions:
 * something like `si1133_add_channel(si1133_t *dev, si1133_sensor_t sensor, uint32_t gain, other channel settings)`
 * or a one-shot  `si1133_set_channels(si1133_t *dev, struct si1133_channel_config_t channels[], size_t nchannels)`
 * or one by one `si1133_set_channel(si1133_t *dev, size_t channel, si1133_sensor_t sensor, uint32_t gain, other channel settings)`

Then as a second step you use one of the following functions to query it:
* `si1133_ret_code_t si1133_capture_sensors(si1133_t *dev, uint32_t values[6])` (implemented here)
* `si1133_ret_code_t si1133_capture_sensors_async(si1133_t *dev, void(cb*)(uint32_t values[6]))` (in a follow up) which will trigger an interrupt when the data is ready, ideally allowing the CPU to sleep for 1~2ms.
* `si1133_ret_code_t si1133_capture_sensors_periodic(si1133_t *dev, void(cb*)(uint8_t channel_mask, uint32_t values[6]))` (in a follow up) which keeps calling the callback every X time, based on the sensor config. Note that in this case what sensors were measured is a bit more messy because each channel is sampled every `800us * MEASRATE * MEASCOUNT[COUNTER_INDEX[channel]]` where `MEASRATE` is global and 16-bit (so, from 800us rate to ~54s rate) and each channel can select one of 3 different global MEASCOUNT[] which is an 8-bit counter; so different can be sampled at different rates; but not all arbitrary.

I wanted to move some of this complexity in handling the sensor into the driver rather than giving the user all the power to use this sensor but also exposing all these issues. If you only care about force-reads like with the saul interface, this could be simple, but we also need to take into account these other use cases.

WDYT?

Reply:
I reimplemented the configuration method, not you have more control over the channels which will help when setting up continuous and interrupt modes. I also made it wait (block) the right amount of time and only have a short timeout when waiting for the command to finish.